jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -427,8 +475,14 @@ def mos_niriss_parser(app, data_dir, obs_label=""):
 
         pupil_id_dict[pupil] = {}
 
-        for row in parsed_cat_fields:
-            pupil_id_dict[pupil][int(row[0])] = (row[1], row[2])
+        if len(cat_fields) == 2: # new style
```

Comment:
So, are the source catalogues allowed to be mixed among "new" and "old" styles?

* If yes: `cat_fields` should be recomputed at every iteration here.
* If no: There is no need to recompute `len(cat_fields)` at every iteration.

Reply:
I tried to support both scenarios because I don't know if the old one is no longer valid. Whom should I ask about that?

Reply:
🤷 I'll have to defer to @javerbukh or @rosteen .

Reply:
This question doesn't apply anymore given the resolution of the conversation just before this one. I would instead go to the questions in the pull request's initial post.